modules/features/player/src/main/java/au/com/shiftyjelly/pocketcasts/player/view/shelf/ShelfBottomSheetPage.kt

```diff
@@ -0,0 +1,175 @@
+            onEditButtonClick = {
+                shelfViewModel.onEditButtonClick()
+                onEditButtonClick()
+            },
+            mediaRouteButton = {
+                MediaRouteButton(
+                    performClick = performMediaRouteButtonClick,
+                    onMediaRouteButtonClick = {
+                        shelfViewModel.onMediaRouteButtonClick()
+                    },
+                )
+            },
+        ) {
+            MenuShelfItems(
+                shelfViewModel = shelfViewModel,
+                onClick = { item, enabled ->
+                    coroutine.launch {
+                        if (item == ShelfItem.Cast) {
+                            performMediaRouteButtonClick = true
+                            delay(100) // allow perform action to complete before dismissing the bottom sheet
+                        }
+                        onShelfItemClick(item, enabled)
+                    }
+                },
+            )
+        }
+    }
+}
+
+@Composable
+private fun Content(
+    onEditButtonClick: () -> Unit,
+    mediaRouteButton: @Composable () -> Unit,
+    menuShelfItems: @Composable () -> Unit,
+) {
+    Column(
+        modifier = Modifier
+            .fillMaxWidth(),
+    ) {
+        Spacer(Modifier.height(8.dp))
+
+        Pill(
+            modifier = Modifier
+                .align(Alignment.CenterHorizontally)
+                .background(MaterialTheme.theme.colors.playerContrast01),
+        )
+
+        Spacer(Modifier.height(8.dp))
+
+        mediaRouteButton()
+
+        Row(
+            verticalAlignment = Alignment.CenterVertically,
+            horizontalArrangement = Arrangement.SpaceBetween,
+            modifier = Modifier
+                .fillMaxWidth()
+                .padding(start = 16.dp),
+        ) {
+            TextH30(
+                text = stringResource(LR.string.player_more_actions),
+                color = MaterialTheme.theme.colors.playerContrast01,
+            )
+
+            IconButton(
+                onClick = onEditButtonClick,
+            ) {
+                Icon(
+                    painter = painterResource(IR.drawable.ic_edit),
+                    contentDescription = stringResource(LR.string.edit),
+                    tint = MaterialTheme.theme.colors.playerContrast01,
+                )
+            }
+        }
+        menuShelfItems()
+    }
+}
+
+@Composable
+private fun MediaRouteButton(
+    performClick: Boolean,
+    onMediaRouteButtonClick: () -> Unit,
+) {
+    AndroidView(
+        factory = { context ->
+            MediaRouteButton(context).apply {
+                visibility = View.GONE
+                setOnClickListener {
+                    onMediaRouteButtonClick()
+                }
+                CastButtonFactory.setUpMediaRouteButton(context, this)
+            }
+        },
+        update = { view ->
+            if (performClick) {
+                view.performClick()
+            }
```

Comment:
This is error-prone to recomposition. It will trigger on every recomposition meaning that `performClick()` can be called multiple times as long as `performClick` input is `true`.

Reply:
Is it ok to unset `performClick` like in fcc3ae4 or do you recommend something better?

Reply:
I don't think this is the right approach. It still can lead to multiple calls in recomposition. While it might seem like an overkill I think something like this should be correct. There might be better approaches as I'm thinking on my feet here but the general idea should be ok.

```diff
diff --git a/modules/features/player/src/main/java/au/com/shiftyjelly/pocketcasts/player/view/shelf/ShelfBottomSheetPage.kt b/modules/features/player/src/main/java/au/com/shiftyjelly/pocketcasts/player/view/shelf/ShelfBottomSheetPage.kt
index afd3778b2..c1476f0d9 100644
--- a/modules/features/player/src/main/java/au/com/shiftyjelly/pocketcasts/player/view/shelf/ShelfBottomSheetPage.kt
+++ b/modules/features/player/src/main/java/au/com/shiftyjelly/pocketcasts/player/view/shelf/ShelfBottomSheetPage.kt
@@ -41,6 +41,7 @@ import au.com.shiftyjelly.pocketcasts.preferences.model.ShelfItem
 import au.com.shiftyjelly.pocketcasts.ui.theme.Theme
 import com.google.android.gms.cast.framework.CastButtonFactory
 import kotlinx.coroutines.delay
+import kotlinx.coroutines.flow.MutableSharedFlow
 import kotlinx.coroutines.flow.map
 import kotlinx.coroutines.launch
 import au.com.shiftyjelly.pocketcasts.images.R as IR
@@ -57,7 +58,7 @@ fun ShelfBottomSheetPage(
         playerViewModel.trimmedShelfLive.asFlow()
             .map { it.copy(it.first.drop(4), it.second) }
     }.collectAsStateWithLifecycle(null)
-    var performMediaRouteButtonClick by remember { mutableStateOf(false) }
+    val performMediaRouteClick = remember { MutableSharedFlow<Unit>() }
     val coroutineScope = rememberCoroutineScope()
     trimmedShelf?.let { (shelfItems, episode) ->
         Content(
@@ -67,10 +68,9 @@ fun ShelfBottomSheetPage(
             },
             mediaRouteButton = {
                 MediaRouteButton(
-                    performClick = performMediaRouteButtonClick,
+                    clickTrigger = performMediaRouteClick,
                     onMediaRouteButtonClick = {
                         shelfViewModel.onMediaRouteButtonClick()
-                        performMediaRouteButtonClick = false
                     },
                 )
             },
@@ -80,7 +80,7 @@ fun ShelfBottomSheetPage(
                 onClick = { item, enabled ->
                     coroutineScope.launch {
                         if (item == ShelfItem.Cast) {
-                            performMediaRouteButtonClick = true
+                            performMediaRouteClick.emit(Unit)
                             delay(100) // allow perform action to complete before dismissing the bottom sheet
                         }
                         onShelfItemClick(item, enabled)
@@ -144,9 +144,10 @@ private fun Content(
 
 @Composable
 private fun MediaRouteButton(
-    performClick: Boolean,
+    clickTrigger: MutableSharedFlow<Unit>,
     onMediaRouteButtonClick: () -> Unit,
 ) {
+    val scope = rememberCoroutineScope()
     AndroidView(
         factory = { context ->
             MediaRouteButton(context).apply {
@@ -155,11 +156,10 @@ private fun MediaRouteButton(
                     onMediaRouteButtonClick()
                 }
                 CastButtonFactory.setUpMediaRouteButton(context, this)
-            }
-        },
-        update = { view ->
-            if (performClick) {
-                view.performClick()
+
+                scope.launch {
+                    clickTrigger.collect { performClick() }
+                }
             }
         },
     )
```

Reply:
Thanks! Applied in a5246b1.